packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -137,12 +138,16 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         _electionStore().nextNominationPeriodPercent = percent;
     }
 
-    function elect(address[] memory candidates) external override {
+    function elect(address[] memory sortedCandidates, uint[] memory positions) external override {
         ElectionStore storage store = _electionStore();
         ElectionData storage electionData = store.electionData;
 
-        if (candidates.length == 0 || candidates.length > electionData.nominees.length) {
-            revert InvalidCandidatesCount();
+        if (
+            sortedCandidates.length == 0 ||
+            sortedCandidates.length > electionData.nominees.length ||
+            sortedCandidates.length != positions.length
+        ) {
+            revert InvalidCandidatesAddresses();
```

Comment:
I think I'd prefer to split these into individual checks with more explicit errors.
`TooFewCandidates`, `TooManyCandidates`, `CandidateLengthMismatch`, etc

Reply:
Done, with `TooManyCandidates`, `MissingCandidates` & `CandidateLengthMismatch`